File: RQ3/Java-REval-Desc/task75_desc.java
public class Solution {
    public static void specialFilter(int[] nums) {
        int count = 0;
        for (int num : nums) {
            if (num > 10) {
                int[] odd_digits = {1, 3, 5, 7, 9};
                String number_as_string = Integer.toString(num);
                if (contains(odd_digits, Character.getNumericValue(number_as_string.charAt(0))) 
                    && contains(odd_digits, Character.getNumericValue(number_as_string.charAt(number_as_string.length() - 1)))) {
                    count += 1;
                }
            }
        }
        int _result = count;

        assert func(_result); // POST: func is equivalent to: Write a function that takes an array of numbers as input and returns the number of elements in the array that are greater than 10 and both first and last digits of a number are odd (1, 3, 5, 7, 9). For example: specialFilter([15, -73, 14, -15]) => 1 specialFilter([33, -2, -3, 45, 21, 109]) => 2
    }
}

File: RQ3/C-REval-Desc/task75_desc.c
#include <stdio.h>
#include <string.h>
#include <stdlib.h>

int func(int result);

int specialFilter(int nums[], int size) {
    int count = 0;
    int odd_digits[] = {1, 3, 5, 7, 9};

    for (int i = 0; i < size; i++) {
        int num = nums[i];
        if (num > 10) {
            char number_as_string[12];
            sprintf(number_as_string, "%d", num);
            if (strchr("13579", number_as_string[0]) && strchr("13579", number_as_string[strlen(number_as_string) - 1])) {
                count++;
            }
        }
    }

    int _result = count;

    assert(func(_result));

    return _result;
}